tests/integration/awslambda/test_lambda.py

```diff
@@ -377,6 +378,99 @@ def test_runtime_introspection_arm(self, lambda_client, create_lambda_function,
         invoke_result = lambda_client.invoke(FunctionName=func_name)
         snapshot.match("invoke_runtime_arm_introspection", invoke_result)
 
+    @pytest.mark.skipif(is_old_provider(), reason="unsupported in old provider")
+    @pytest.mark.skipif(
+        not use_docker(),
+        reason="Monkeypatching of Docker-related flag not applicable if run locally",
+    )
```

Comment:
Why wouldn't this work?

Reply:
Ah, now i understand. The `if run locally` confused me, as I thought of running the test locally, not running lambdas in the local executor. Overlooked the `use_docker` and thought of `in_docker`.

Still, use_docker might be problematic, as it would skip the tests if I set (by accident) something weird like `LAMBDA_EXECUTOR=local PROVIDER_OVERRIDE_LAMBDA=v2`.

Reply:
I clarified the reason and added a new `is_old_local_executor()` util. This skip is currently redundant for `test_ignore_architecture` given the old provider is unsupported anyways but needs to be extended to the new local executor replacement (i.e., the self-managed workers) because they do not spawn new containers either.